client/packages/common/src/ui/discovery/DiscoveredServers.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import {
+  AlertIcon,
+  Box,
+  HomeIcon,
+  InlineSpinner,
+  ListItemIcon,
+  ListItemText,
+  MenuItem,
+  MenuList,
+  Typography,
+  useTranslation,
+  frontEndHostDisplay,
+  CheckboxEmptyIcon,
+  FrontEndHost,
+  useNativeClient,
+  GqlProvider,
+  QueryClientProvider,
+  QueryClient,
+  frontEndHostGraphql,
+  useInitialisationStatus,
+  InitialisationStatusType,
+} from '@openmsupply-client/common';
+
+type ConnectToServer = ReturnType<typeof useNativeClient>['connectToServer'];
+
+type DiscoverServersProps = {
+  servers: FrontEndHost[];
+  connect: ConnectToServer;
+  discoveryTimedOut: boolean;
+};
+
+export const DiscoveredServers = ({
+  servers,
+  connect,
+  discoveryTimedOut,
+}: DiscoverServersProps) => {
+  const t = useTranslation('app');
+
+  if (servers.length === 0)
+    return (
+      <Box
+        display="flex"
+        flex={1}
+        justifyContent="center"
+        alignContent="center"
+      >
+        <InlineSpinner messageKey="searching" />
+      </Box>
+    );
+
+  if (discoveryTimedOut)
+    return (
+      <Box
+        display="flex"
+        sx={{ color: 'error.main' }}
+        justifyContent="center"
+        flexDirection="column"
+      >
+        <Box display="flex" gap={1}>
+          <Box>
+            <AlertIcon />
+          </Box>
+          <Box>
+            <Typography sx={{ color: 'inherit' }}>
+              {t('error.server-not-found')}
+            </Typography>
+          </Box>
+        </Box>
+      </Box>
+    );
+
+  return (
+    <Box sx={{ minWidth: '300px', color: 'gray.dark' }}>
+      <Typography
+        sx={{
+          fontSize: {
+            xs: '19px',
+            sm: '19px',
+            md: '24px',
+            lg: '32px',
+            xl: '32px',
+          },
+          fontWeight: 700,
+          color: 'primary.main',
+        }}
+      >
+        {t('discovery.select-server')}
+      </Typography>
+      <MenuList style={{ overflowY: 'auto', maxHeight: '200px' }}>
+        {servers.map(server => (
+          <DiscoveredServerWrapper
+            key={frontEndHostGraphql(server)}
```

Comment:
I'm assuming that `frontEndHostGraphql` was used for convenience? Just needs to be something that uniquely identifies the server right?
Would be nice to have an `id` generated for the `FrontEndHost` which is a combination of hardwareId and port, which then could be used here, and to uniquely identify servers to connect to

Reply:
sounds good, I hope inline should be ok for now: 109fa835e25ab63fffb7797b0c0f8ad3d70dd8e9